### DISCUSSION TITLE
I cannot delete the user from the auth.users table

### DISCUSSION THREAD

**Discussion by Fistonkul** [supabase/supabase#13302] [Questions]
I have tried to alter it and I have dropped the rows from it and now it is now allowing me to sign in anymore , any help

**Comment by GaryAustin1** [comment]
You are not providing much info.
Alter it how?
What errors are you getting now?
You can look in the Postgres logs of the dashboard for possible errors.

**Reply by Fistonkul** [reply]
Hi sorry, I tried to use "ALTER TABLE auth.users DROP COLUMN name;" on auth.users table and then drop the column name 

and as result when I tried to delete the user from the table in the dashboard it says "Failed to delete user: Database error loading user", I don't  how to reset this , and I cannot sign in anymore into my account

**Reply by GaryAustin1** [reply]
That is a pretty bad thing to do.  Depending on what the column was and if you had any user data in it, you might be able to add the column back or not.  
You might be better just restarting a new instance if you were just starting.

**Reply by Fistonkul** [reply]
I am just deleting the three users that I have now from SQL query , hope it works

**Reply by Fistonkul** [reply]
I have deleted all the user from the auth.user now I cannot sign up new users , what can I do to fix this

**Reply by GaryAustin1** [reply]
You said you dropped a column or columns in auth.users.  You would have to put them back exactly like they were with any constraints they may have had on them.  The command you used did not just remove data from a column, it dropped the column and all of its settings from the table.

It might be easier to start a new instance, versus trying to fix the auth.users table, depending on what you did to it.

**Reply by Fistonkul** [reply]
what do you mean by starting a new instance ?

i have no user in auth users table now , I was trying to sign up again it throws this " Signup error: AuthApiError: Database error finding user"

the code still the same , and the all the tables still the same

**Reply by GaryAustin1** [reply]
I thought you said you ran "ALTER TABLE auth.users DROP COLUMN name;" ?  If you did that you have to add the column/s back to auth.users table exactly of the type they were.
If you did not do that then I don't know what could be causing your database error.
You can look in Postgres log to see what the real error is.   (Likely an error about a missing column in auth.users table).

I'm saying if you damaged the auth.users table it is probably easier to create a new Supabase instance and start over if you are just beginning.

**Reply by Fistonkul** [reply]
this is frustrating now, where can i find Postgres log, 

I had the auth table with email and I tried to alter it with adding name and phone from query window and drop them after and after I could not login , i deleted all the users from the auth tables , i want to restart sign up users , but it does not work ,

**Reply by GaryAustin1** [reply]
I'm sorry you are frustrated.  You are only providing words for what you did to the auth.users table.  You should never modify any tables in Supabase schemas.  I can't guess what columns you added and then deleted or what the state of your auth.users tables is now (if all of the columns are back).

Postgres log MIGHT tell you what column is missing still in auth.users, if that is your problem...
![image](https://user-images.githubusercontent.com/54564956/228682748-f47df14c-e1df-43b5-9924-346f24248c31.png)

You may need to contact support, if you can't just start over with a new instance at this point.

**Reply by Fistonkul** [reply]
can I regenrate a new api key , will it fix it , i think the database is broken , especially the auth users

**Reply by GaryAustin1** [reply]
If you altered auth.user table, then yes you likely broke it.
A new API key will not help.
You basically have to create a new instance from scratch.
If you have alot of data you might be able to migrate it from your current instance to the new:  https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects#migrate-your-project

Once again support MIGHT be able to help.

**Reply by Fistonkul** [reply]
yea thanks , i have just five tables so its not huge 

I wont need to change the code right , once I have the new instance , I can API keys and it will work right

**Reply by GaryAustin1** [reply]
Your client code is not impacted, besides the URL and anon key.

**Reply by Fistonkul** [reply]
Hi, I have the the auth users table and public users table that store the user 's email and I want to delete the user account from the client side but it is not allowing me to delete and permission denied  , I have this function 
 BEGIN
  DELETE FROM public.users WHERE id = user_id;
  DELETE FROM auth.users WHERE id = user_id;
END;

and I have grant permission to all the user to delete but it is not deleting it, any help would be appreciated

**Reply by GaryAustin1** [reply]
This seems to be a suggested way...
https://github.com/orgs/supabase/discussions/1066#discussioncomment-1601641

**Reply by Fistonkul** [reply]
Hi , I m trying to reset a password in supabase but it is not working at all , is supabase reset password free
